common/schema/edges.go

```diff
@@ -0,0 +1,81 @@
+package schema
+
+import (
+	"slices"
+	"strings"
+)
+
+// GraphNode provides inbound and outbound edges for a node.
+type GraphNode struct {
+	Decl Decl
+	In   []RefKey
+	Out  []RefKey
+}
+
+// Graph returns a biderectional graph representation of the schema.
+func Graph(s *Schema) map[RefKey]GraphNode {
+	// Build graph with outbound edges.
+	result := map[RefKey]GraphNode{}
+
+	for _, module := range s.Modules {
+		Visit(module, func(s Node, next func() error) error { //nolint:errcheck
+			d, ok := s.(Decl)
+			if !ok {
+				return next()
+			}
+
+			// Ignore type parameters.
+			ignoredRefs := map[RefKey]bool{}
+			if data, ok := d.(*Data); ok {
+				for _, tp := range data.TypeParameters {
+					ignoredRefs[RefKey{Name: tp.GetName()}] = true
```

Comment:
Should this have the Module in it?

Reply:
No, `HttpRequest<Body, Path, Query>` has child refs to `Body`, `Path` and `Query` which is what we are filtering out here and they don't have a module name.